### PR TITLE
Fix two issues with Federation log dumping in e2e.go

### PR DESCRIPTION
**Pull request by perotinus**
- Add a missing check that the test being run is a Federation test
- Check for the presence of federation/cluster/log-dump.sh before running it

Fixes the failure in #44477.

### CODE REVIEW

**Comment by perotinus** [comment]
cc @madhusudancs @fejta

**Comment by fejta** [comment]
@k8s-bot ok to test

File: kubetest/e2e.go [hunk 1]
```diff
@@ -392,8 +394,15 @@ func DumpClusterLogs(location string) error {
 }
 
 func DumpFederationLogs(location string) error {
-	log.Printf("Dumping Federation logs to: %v", location)
-	return finishRunning(exec.Command("./federation/cluster/log-dump.sh", location))
+	logDumpPath := "./federation/cluster/log-dump.sh"
+	// federation/cluster/log-dump.sh only exists in the Kubernetes tree
+	// post-1.6. If it doesn't exist, do nothing and do not report an error.
+	if i, err := os.Stat(logDumpPath); err == nil && !i.IsDir() {
```
Comment (madhusudancs) [inline kubetest/e2e.go]: Shouldn't you actually check that this is a regular file `i.Mode().IsRegular()`?

Comment (perotinus) [inline kubetest/e2e.go]: Checking IsDir seemed odd to me too, but I was copying what someone else did elsewhere in this repo: https://github.com/kubernetes/test-infra/blob/72232020680b657f2f37b18bd5f46fa42ece272b/kubetest/extract.go#L234

Not an excuse, but I assumed that there was a good reason for that. @fejta, do you know why?

Comment (fejta) [inline kubetest/e2e.go]: IMO that is unnecessarily specific. For example, what will `i.Mode().IsRegular()` do if `log-dump.sh` is a symlink to the real log dump?

Comment (madhusudancs) [inline kubetest/e2e.go]: Fair question. But on the other hand, if the file is a `SymLink()` there are other problems. The sym link could be broken. It is probably not worth including all possible permutations of checks.

Anyway, I don't feel too strongly about this. It is fine either way.

Comment (fejta) [inline kubetest/e2e.go]: I'd be fine removing the `IsDir()` check. Just prefer to not be overly restrictive.

Comment (perotinus) [inline kubetest/e2e.go]: SGTM. I've removed the check.

PTAL.

**Comment by perotinus** [comment]
Hmm. The failed bazel test passes locally. Is there something else that broke that test?

**Comment by krzyzacy** [comment]
That's not your fault, we are fixing it. PTAL with the latest change and we can merge.

**Comment by fejta** [comment]
@rmmh re bazel test outage

**Comment by fejta** [comment]
@k8s-bot bazel test this

**Pull request merged**